d2bs/kolbot/libs/common/Mercenary.js

```diff
@@ -0,0 +1,459 @@
+
+				var v = this.getVariant();
+				print("my new merc variant is: " + v + " - desired variant: " + args.variant);
+
+				if (args.variant === v) {
+					break;
+				}
+
+			} while (!this.hasMerc(args.mercType, args.variant) && Mercenary.availableMercs.length > 0);
+
+		} else {
+			print("No mercs available");
+
+			me.cancel();
+
+			return false;
+
+		}
+
+		me.cancel();
+
+		return true;
+
+	},
+
+	revive: function () {
+		if (!this.needMerc()) {
+			print("I don't need to revive my merc");
+
+			return true;
+		}
+
+		if (!this.isDead()) {
+			return true;
+		}
+
+		// Fuck Aheara
+		if (me.act === 3) {
+			Town.goToTown(2);
+		}
+
+		var i, tick, dialog, lines,
+			preArea = me.area,
+			npc = Town.initNPC("Merc", "reviveMerc");
+
+		if (!npc) {
+			return false;
+		}
+
+		MainLoop:
+		for (i = 0; i < 3; i += 1) {
+			dialog = getDialogLines();
+
+			for (lines = 0; lines < dialog.length; lines += 1) {
+				if (dialog[lines].text.match(":", "gi")) {
+					dialog[lines].handler();
+					delay(Math.max(750, me.ping * 2));
+				}
+
+				// "You do not have enough gold for that."
+				if (dialog[lines].text.match(getLocaleString(3362), "gi")) {
+					return false;
+				}
+			}
+
+			while (getTickCount() - tick < 2000) {
+				if (this.getMerc()) {
+					delay(Math.max(750, me.ping * 2));
+
+					break MainLoop;
+				}
+
+				delay(200);
+			}
+		}
+
+		Attack.checkInfinity();
+
+		if (this.getMerc()) {
+			if (Config.MercWatch && !me.inTown) { // Cast BO on merc so he doesn't just die again
+				print("MercWatch precast");
+				Pather.useWaypoint("random");
+				Precast.doPrecast(true);
+				Pather.useWaypoint(preArea);
+			}
+
+			return true;
+		}
+
+		return false;
+
+	},
+
+	unloadEquipment: function () {
+		print("Unloading merc equipment");
+		var cursorItem;
+
+		//ok this is a bit stupid
+		clickItem(4, 4);
+		delay(me.ping + 500);
+
+		if (me.itemoncursor) {
+			delay(me.ping + 500);
+			cursorItem = getUnit(100);
+
+			if (cursorItem) {
+				if (!Storage.Inventory.MoveTo(cursorItem)) {
+					return false;
+				}
+
+				delay(me.ping + 1000);
+
+				if (me.itemoncursor) {
+					Misc.click(0, 0, me);
+					delay(me.ping + 500);
+				}
+			}
+		}
+
+		clickItem(4, 3);
+		delay(me.ping + 500);
+
+		if (me.itemoncursor) {
+			delay(me.ping + 500);
+			cursorItem = getUnit(100);
+
+			if (cursorItem) {
+				if (!Storage.Inventory.MoveTo(cursorItem)) {
+					return false;
+				}
+
+				delay(me.ping + 1000);
+
+				if (me.itemoncursor) {
+					Misc.click(0, 0, me);
+					delay(me.ping + 500);
+				}
+
+			}
+
+		}
+
+		clickItem(4, 1);
+		delay(me.ping + 500);
+
+		if (me.itemoncursor) {
+			delay(me.ping + 500);
+			cursorItem = getUnit(100);
+
+			if (cursorItem) {
+				if (!Storage.Inventory.MoveTo(cursorItem)) {
+					return false;
+				}
+
+				delay(me.ping + 500);
+
+				if (me.itemoncursor) {
+					Misc.click(0, 0, me);
+					delay(me.ping + 500);
+				}
+			}
+		}
+
+		return true;
+
+	},
+
+	_gamePacket: function (bytes) {

```

Comment:
rename this to something descriptive specific to this lib
refactor to something like:
```
if (bytes[0] !== 0x4e) {
    return true;
}

var id = (bytes[2] << 8) + bytes[1];

if (Mercenary.availableMercs.indexOf(id) !== -1) {
    Mercenary.availableMercs.length = 0;
}

Mercenary.availableMercs.push(id);
return true;
```